Client/WebFilters/AdblockResourceDownloader.swift

```diff
@@ -46,91 +46,92 @@ class AdblockResourceDownloader {
     if now.timeIntervalSince(lastFetchDate) >= fetchInterval {
       lastFetchDate = now
 
-      AdblockResourceDownloader.shared.regionalAdblockResourcesSetup()
-      AdblockResourceDownloader.shared.generalAdblockResourcesSetup()
+      Task.detached(priority: .userInitiated) {
+        do {
+          try await withThrowingTaskGroup(of: Void.self) { [weak self] group in
+            guard let self = self else { return }
+            group.addTask { try await self.regionalAdblockResourcesSetup() }
+            group.addTask { try await self.generalAdblockResourcesSetup() }
+            try await group.waitForAll()
+          }
+        } catch {
+          log.error("Failed to Download Adblock-Resources: \(error)")
+        }
+      }
     }
   }
 
-  func regionalAdblockResourcesSetup() {
+  func regionalAdblockResourcesSetup() async throws {
     if !Preferences.Shields.useRegionAdBlock.value {
       log.debug("Regional adblocking disabled, aborting attempt to download regional resources")
       return
     }
 
-    downloadResources(
+    try await downloadResources(
       type: .regional(locale: locale),
-      queueName: "Regional adblock setup"
-    ).uponQueue(.main) {
-      log.debug("Regional blocklists download and setup completed.")
-      Preferences.Debug.lastRegionalAdblockUpdate.value = Date()
-    }
+      queueName: "Regional adblock setup")
+    log.debug("Regional blocklists download and setup completed.")
+    Preferences.Debug.lastRegionalAdblockUpdate.value = Date()
   }
 
-  func generalAdblockResourcesSetup() {
-    downloadResources(
+  func generalAdblockResourcesSetup() async throws {
+    try await downloadResources(
       type: .general,
-      queueName: "General adblock setup"
-    ).uponQueue(.main) {
-      log.debug("General blocklists download and setup completed.")
-      Preferences.Debug.lastGeneralAdblockUpdate.value = Date()
-    }
+      queueName: "General adblock setup")
+    log.debug("General blocklists download and setup completed.")
+    Preferences.Debug.lastGeneralAdblockUpdate.value = Date()
   }
 
-  private func downloadResources(type: AdblockerType, queueName: String) -> Deferred<()> {
-    let completion = Deferred<()>()
-
-    let queue = DispatchQueue(label: queueName)
+  private func downloadResources(type: AdblockerType, queueName: String) async throws {
     let nm = networkManager
     let folderName = AdblockResourceDownloader.folderName
 
     // file name of which the file will be saved on disk
     let fileName = type.identifier
```

Comment:
Is this `type.identifier` coming from the network?

Reply:
It does not. It comes from a hardcoded type like `.dat` or `.json`.